### ISSUE TITLE
Implement Health Checks for Zookeeper and Kafka in docker-compose.yaml

### ISSUE THREAD

**Issue by Celebrinborn** [Celebrinborn/SecurityCamera#36]
To enhance the reliability and startup process of our Docker Compose setup, we need to add health service checks for Zookeeper and Kafka. This will ensure services dependent on these are started only after they are fully operational.

### Acceptance Criteria
1. **Health Checks Implemented:** Health checks are implemented for Zookeeper and Kafka in the `docker-compose.yaml` file.
2. **Dependent Services Configuration:** Services dependent on Zookeeper and Kafka use the `depends_on` option with `condition: service_healthy` to ensure they start only after these services are healthy. Example:
   ```yaml
   services:
     myservice:
       depends_on:
         zookeeper:
           condition: service_healthy
         kafka:
           condition: service_healthy
   ```
3. **Timeout Prevention:** The solution should prevent timeouts during Kafka's boot process.
4. **Documentation:** Include comments in the `docker-compose.yaml` file explaining the health check implementation.

### Difficulty
Medium

### Time Involvement
Estimated time to complete: 3-4 hours

### Implementation References
1. **Netcat for Health Checks:** Use netcat to check if Kafka and Zookeeper services are listening on their respective ports (Kafka: 9092, Zookeeper: 2181). Example command for Kafka: `nc -z localhost 9092 || exit -1`.
2. **Kafka Functionality Check:** As an alternative for Kafka, use a command to describe a Kafka topic to ensure functionality, e.g., `["CMD-SHELL", "kafka-topics.sh --bootstrap-server 127.0.0.1:9092 --topic <TOPIC_NAME> --describe"]`.
3. **Configuration of `depends_on`:** Configure `depends_on` in `docker-compose.yml` with `condition: service_healthy`.
```

This format should render correctly when you create a new issue in your GitHub repository. Let me know if you need any adjustments or additional information!
